### PR TITLE
Fix Linux support

### LINKED ISSUE #1140: Erroneously passing lib and .so decorations to find_library on Linux
Question about library singleton.py. It looks like we recently changed it to pass the DLL name through find_library ([here](https://github.com/ni/nimi-python/commit/54ab723c78415ca90ca3986b7c2a91f588e268d3)), but the DLL names contain "lib" and ".so" -fixes ([here](https://github.com/ni/nimi-python/blob/master/generated/nidmm/_library_singleton.py#L15)), but find_library does not like those prefixes ([here](https://docs.python.org/3/library/ctypes.html#finding-shared-libraries)). This appears to be causing an issue on @Fladolcetta's machine with DMM on OpenSUSE, should we put together a PR for that?

### PR DESCRIPTION
**Pull request by lkolbly**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

- [ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.

(unknown)

- [ ] ~~I've added tests applicable for this pull request~~

### What does this Pull Request accomplish?

Does not pass libraries with lib/.so trimmings to ctypes find_library.

### List issues fixed by this Pull Request below, if any.

* Fix #1140 

### What testing has been done?

Tested that modifying the generated file on a linux machine fixes the crash outlined in the issue.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -440,10 +438,10 @@ Driver specific installation instructions can be found on Read The Docs:
 * `nidcpower <http://nimi-python.readthedocs.io/en/master/nidcpower.html#installation>`_
 * `nidmm <http://nimi-python.readthedocs.io/en/master/nidmm.html#installation>`_
 * `nifgen <http://nimi-python.readthedocs.io/en/master/nifgen.html#installation>`_
-* `nimodinst <http://nimi-python.readthedocs.io/en/master/nimodinst.html#installation>`_
 * `niscope <http://nimi-python.readthedocs.io/en/master/niscope.html#installation>`_
-* `nise <http://nimi-python.readthedocs.io/en/master/nise.html#installation>`_
 * `niswitch <http://nimi-python.readthedocs.io/en/master/niswitch.html#installation>`_
+* `nise <http://nimi-python.readthedocs.io/en/master/nise.html#installation>`_
+* `nimodinst <http://nimi-python.readthedocs.io/en/master/nimodinst.html#installation>`_
```
Comment (marcoskirsch) [inline README.rst]: These are all unrelated changes from previous PRs. Please merge latest master.

File: src/nifgen/metadata/config.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
     'library_info': {
         'Linux': {
             '64bit': {
-                'name': 'libfgen.so',
+                'name': 'fgen',
```
Comment (marcoskirsch) [inline src/nifgen/metadata/config.py]: Please make this `nifgen`. Even though that driver has no Linux support, I am fairly sure the library would not be "fgen".

File: README.rst [hunk 3]
```diff
@@ -45,14 +45,12 @@ About
 The **nimi-python** repository generates Python bindings (Application Programming Interface) for interacting with the Modular Instrument drivers. Currently, the following drivers are supported:
 
 * NI-DCPower (Python module: nidcpower)
-* NI-Digital Pattern Driver (Python module: nidigital)
```
Comment (sbethur) [inline README.rst]: @marcoskirsch, changes in this file is because I updated README without realizing that it's codegen'd. I'm looking into how to update this file the correct way.

File: README.rst (hunk 1 above)
Comment (lkolbly) [reply README.rst]: Based on @sbethur 's comment below, I think it is correct to include this change?

Comment (sbethur) [reply README.rst]: @lkolbly, I'm creating a PR to fix this. Please exclude this file from your PR.

**Comment by sbethur** [comment]
ok to test

File: README.rst (hunk 1 above)
Comment (lkolbly) [reply README.rst]: @sbethur Pushed a commit that should revert it.

**Comment by texasaggie97** [comment]
ok to test
